theano/gof/type.py

```diff
@@ -35,6 +35,19 @@ class CLinkerType(CLinkerObject):
 
     """
 
+    def c_element_type(self):
+        """
+        Optional: Return the name of the primitive C type of items into variables
+        handled by this type.
+
+        e.g:
+
+         - For ``TensorType(dtype='int64', ...)``: should return ``"npy_int64"``.
+         - For ``GpuArrayType(dtype='int32', ...)``: should return ``"ga_int"``.
+
+        """
+        return ''
```

Comment:
why return an empty string and not MethodNotDefined? I think by default, it should be an error is we look for the element type and it isn't provided.